sound/soc/intel/common/soc-acpi-intel-bxt-match.c

```diff
@@ -6,9 +6,43 @@
  *
  */
 
+#include <linux/dmi.h>
 #include <sound/soc-acpi.h>
 #include <sound/soc-acpi-intel-match.h>
 
+static unsigned long apl_machine_id;
+
+#define APL_TDF8532  1
+
+static int apl_tdf8532_quirk_cb(const struct dmi_system_id *id)
+{
+	apl_machine_id = APL_TDF8532;
+	return 1;
+}
+
+static const struct dmi_system_id apl_table[] = {
+	{
+		.callback = apl_tdf8532_quirk_cb,
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "Intel Corp."),
+			DMI_MATCH(DMI_BOARD_NAME, "Apollolake RVP1A"),
+		},
+	},
+	{},
+};
+
+static struct snd_soc_acpi_mach *apl_quirk(void *arg)
+{
+	struct snd_soc_acpi_mach *mach = arg;
+
+	dmi_check_system(apl_table);
+
+	if (apl_machine_id == APL_TDF8532)
+		return NULL;
```

Comment:
so this means we will always return NULL and we can never match with INT34C3?

Reply:
on RVP this ID will not result in any matching for this INT34C3 so the code will keep looking for other IDs and eventually fall to nocodec or HDaudio.
No change on GP-MRB.